x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connector_detail/connector_name_and_description.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { ChangeEvent, useEffect, useState } from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import { EuiFlexGroup, EuiFlexItem, EuiInlineEditText, EuiInlineEditTitle } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import { Connector } from '@kbn/search-connectors';
+
+import { ConnectorNameAndDescriptionLogic } from './connector_name_and_description_logic';
+
+export interface ConnectorNameAndDescriptionProps {
+  connector: Connector;
+}
+
+export interface ResolverObject {
+  rej: (value: boolean | void | PromiseLike<boolean | void>) => void;
+  res: (value: boolean | void | PromiseLike<boolean | void>) => void;
+}
+
+let promise: Promise<boolean | void> | undefined;
+
+const getValidationPromiseResolvers = (): ResolverObject => {
+  const resolvers = {
+    rej: () => {},
+    res: () => {},
+  };
+  promise = new Promise((resolve, reject) => {
+    resolvers.res = resolve;
+    resolvers.rej = reject;
+  });
+  return resolvers;
+};
+
+export const ConnectorNameAndDescription: React.FC<ConnectorNameAndDescriptionProps> = ({
+  connector,
+}) => {
+  const [resolverObject, setResolverObject] = useState<ResolverObject>({
+    rej: () => {},
+    res: () => {},
+  });
+  const [connectorName, setConnectorName] = useState<string>(connector.name);
+  const [connectorDescription, setConnectorDescription] = useState<string | null>(
+    connector.description
+  );
+  const [nameErrors, setNameErrors] = useState<string[]>([]);
+  const { saveNameAndDescription, setConnector } = useActions(ConnectorNameAndDescriptionLogic);
+  const { status, isLoading, isFailed, isSuccess } = useValues(ConnectorNameAndDescriptionLogic);
+  useEffect(() => {
+    setConnector(connector);
+  }, [connector]);
+
+  useEffect(() => {
```

Comment:
this feels like it's overly complicated but I'm not sure what's happening so it's hard to suggest a better way 🤔

Reply:
Issue is the inline edit expects a promise, and to hook it with redux/kea, there is no better solution I have figured. New JS spec has slightly better Promise api, but not available in Safari :/ We should ask eui to ask a different solution.